okhttp/src/jvmMain/kotlin/okhttp3/Cookie.kt

```diff
@@ -326,25 +361,36 @@ class Cookie private constructor(
       this.httpOnly = true
     }
 
+    fun sameSite(sameSite: String) = apply {
+      require(sameSite.trim() == sameSite) { "sameSite is not trimmed" }
+      this.sameSite = sameSite
+    }
+
     fun build(): Cookie {
+      if (sameSite.equals("None", ignoreCase = true)) {
+        require(secure) { "Secure attribute is required for this SameSite policy" }
+      }
```

Comment:
```suggestion
```

I’m anxious about this 'cause I’d like this to work for all cookies:

```kotlin
val cookie: Cookie = ...
val copy = cookie.newBuilder().build()
```

In this case we’re strict on `build()` in a way that we can’t / shouldn’t be on `parse()`.